source/extensions/filters/http/cors/cors_filter.cc

```diff
@@ -125,6 +131,12 @@ Http::FilterHeadersStatus CorsFilter::decodeHeaders(Http::RequestHeaderMap& head
     response_headers->setInline(access_control_max_age_handle.handle(), maxAge());
   }
 
+  if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.cors_private_network_access")) {
+    if (headers.getInlineValue(access_control_request_private_network_handle.handle()) == "true") {
+      response_headers->setInline(access_control_response_private_network_handle.handle(), "true");
+    }
+  }
```

Comment:
I didn't get why this runtime guard is necessary 🤔 If it's necessary indeed, can you add some more comments here?

Reply:
Since I am not sure to add it to API, so add runtime guard here.

Reply:
IMO, we should add a new field in the `envoy.config.route.v3.CorsPolicy` for this new feature. cc @envoyproxy/api-shepherds for some more inputs.

Reply:
Sure.